Added footer and case studies page

DevTaehong opened the pull request:
- Added footer and case studies page
- Restructured SVG components
- Restructured folders

![Screenshot 2023-09-14 at 3 57 19 PM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/536136d1-6a83-4c04-8916-5020aebc1ba6)
![Screenshot 2023-09-14 at 3 56 59 PM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/8314d287-b609-4b1d-a5d5-2668a76b5688)

Nassim-Saboundji commented on components/CaseStudiesHero.tsx:
```diff
@@ -0,0 +1,31 @@
+              />
+              <span className="relative">{text}</span>
+            </span>{" "}
+            {text !== "Studies" && <br className="block sm:hidden" />}
```
You shouldn't use the br element for spacing. You should instead use padding or margins to create the spacing you need.

Nassim-Saboundji reviewed (changes requested):
Good job!

I noticed that there is a lot of code to handle svg states (onHover, etc...) that is repeated from one svg to the next. Can you look into how you could refactor your code so you don't have to this much repetition.

cdsalty commented:
@DevTaehong You are still not rendering any text when the user is on 'light mode.' Could you be sure to correct this as well?
<img width="765" alt="Screen Shot 2023-09-14 at 7 19 44 PM" src="https://github.com/DevTaehong/portfolio_jsm/assets/33855399/92e94c41-b458-47e5-80f3-f02575c74523">

DevTaehong commented:
> @DevTaehong You are still not rendering any text when the user is on 'light mode.' Could you be sure to correct this as well? <img alt="Screen Shot 2023-09-14 at 7 19 44 PM" width="765" src="https://user-images.githubusercontent.com/33855399/268131768-92e94c41-b458-47e5-80f3-f02575c74523.png">

My bad :( forgot to change one variable. I've made sure it works.

DevTaehong commented:
@Nassim-Saboundji @cdsalty I've made the requested changes. Thanks for the feedback. the code looks much better.

cdsalty commented on components/navBar/NavBarMenuButton.tsx:
```diff
@@ -1,4 +1,4 @@
-const NavBarMenuButton = () => {
+function NavBarMenuButton() {
   return (
     <svg
```
Place this inside an SVG and import only the SVG if possible.

DevTaehong replied:
![Screenshot 2023-09-15 at 2 37 25 AM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/0b11669d-e0c4-41c3-83c5-fbb70e179e12)

I'm using it as an icon for NexUI and the icon prop should be ReactNode.

cdsalty commented on components/CaseStudiesHero.tsx:
```diff
@@ -0,0 +1,30 @@
+const heroText = ["Case", "Studies"];
```
Move this inside its own index file along with any other dummy data or hard-coded data.

cdsalty commented on components/CaseStudiesHero.tsx:
```diff
@@ -0,0 +1,30 @@
+      <h1 className="mobileHeading1 max-w-[21rem] dark:text-white800 sm:max-w-[41.125rem] lg:text-[4rem] lg:leading-[5.2rem] lg:tracking-[-0.04rem]">
+        Recent{" "}
+        {heroText.map((text) => (
+          <>
+            <span key={text} className="relative inline-block">
+              <span
+                className={`${
+                  text === "Case" && "sm:w-[15rem]"
+                } absolute top-[1.88rem] h-[0.8125rem] w-full shrink-0 bg-accentSecondary 
+                lg:top-[3.6rem] lg:h-[1.68963rem]`}
+              />
+              <span className="relative">{text}</span>
+            </span>{" "}
+          </>
```
Here are a few things to review:
- Consider making this a reusable component.
- You don't need the fragment, A `<div>` would be more appropriate here. Also, you would have placed your key in the fragment, not the `<span>`
- I understand you don't have much data but it's still good practice to use a unique key when possible. 

```
// projectsData.ts
export const projects = [
  // ...
];

```

```
// Import it in your component
import { projects } from './projectsData';

```

cdsalty commented on components/CaseStudiesProject.tsx:
```diff
@@ -0,0 +1,40 @@
+  project: { name: string; description: string; image: string; color: string };
+}) => {
+  return (
+    <>
```
I don't think the React Fragment is necessary here.

DevTaehong replied:
![Screenshot 2023-09-15 at 2 55 03 AM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/a8b20f46-849a-45dd-835e-344e4619d232)

Without the Fragment, it causes an error. Did you mean to use a div tag instead?

![Screenshot 2023-09-15 at 2 56 25 AM](https://github.com/DevTaehong/portfolio_jsm/assets/71358207/506137c0-dd6a-408a-ad6f-6ff57f95abc8)

cdsalty commented on components/CaseStudiesProject.tsx:
```diff
@@ -0,0 +1,40 @@
+        <Image
+          className="lg:h-[15.8rem] lg:w-[26.8rem]"
+          src={project.image}
+          width={278.486}
+          height={165.054}
```
FYI: The width and height attributes are used to infer the correct aspect ratio of the image and avoid layout shift from the image loading in. The width and height do not determine the rendered size of the image file. Learn more about [Image Sizing](https://nextjs.org/docs/app/building-your-application/optimizing/images#image-sizing).
You can also read more about it here: https://nextjs.org/docs/app/building-your-application/optimizing/images#local-images

cdsalty commented on components/CaseStudiesProjects.tsx:
```diff
@@ -0,0 +1,56 @@
+        xl:flex xl:flex-row xl:flex-wrap xl:px-[5.31rem] 2xl:mx-auto 2xl:max-w-[90rem]"
+      >
+        {projects.map((project) => (
+          <div
+            key={project.name}
+            className="max-w-[21.5625rem] lg:max-w-[34.375rem]"
+          >
+            <CaseStudiesProject project={project} />
+          </div>
+        ))}
+      </div>
```
As mentioned earlier, you could move these into a separate component.

cdsalty commented on components/Footer.tsx:
```diff
@@ -0,0 +1,45 @@
+  LinkedInIcon,
+} from "@/components/svg/footer";
+
+const links = [
+  {
+    link: "https://www.linkedin.com/in/taehong/",
+    icon: LinkedInIcon,
+  },
+  {
+    link: "https://github.com/DevTaehong",
+    icon: FooterGitHubIcon,
+  },
+];
```
Move this data outside the component as well.

cdsalty reviewed (changes requested):
Looking great @DevTaehong! To further align it with our project specifications, please consider the following updates:

Morrent Project Background: Update the background to the blue color specified in the design to enhance section differentiation.

Text Color in Light Mode: The text on the Morrent project card is currently set to white in light mode. Please adjust this for better readability.

Footer Alignment: The footer is well-designed but should be aligned at the bottom of the div as per the design guidelines.

Tablet Responsiveness: Verify the responsiveness of the 'Work Experience' section, particularly when viewed on tablets.

Error Codes: Several errors have been logged in your developer console. Please make sure that these are resolved before your next submission.

Extract SVG to a Separate File
Consider storing the SVG asset in a separate SVG file if it is static and doesn't change. This makes the React component easier to read and manage. You can then import and use it in your React component.


Please review the code suggestions I've provided in your pull request. You're doing excellent work, and by focusing on reusable components and externalizing hardcoded data, we can elevate the project even further.


Could you make sure that the PR contains only the changes mentioned above?

Thank you for your attention to these details, and keep up the great work!

DevTaehong commented:
@cdsalty 
Morrent Project Background: I've made sure that it renders the colours properly according to the design.

Text Color in Light Mode: The blue color is rendering properly so I've applied the text color that the design has.

Footer Alignment: Now it's aligned. Thanks for mentioning it.

Tablet Responsiveness: I've changed it.

Error Codes: To get rid of all errors I had, I've changed all components related to the errors.

Extract SVG to a Separate File: I've only made svg assets React components when the colour needs to be changed in dark mode. Is there another way I can change SVG's colour without making components?

Thank you for your feedback! I was missing a lot of things lol

Nassim-Saboundji reviewed (approved):
Looks good!

When it comes to SVGs. I usually recommended using [SVGR](https://react-svgr.com/) which allows you to import svgs as React components that are optimized. Now for modifying the colors you need to jumps through some hoops but at least you don't need to create multiple svgs of different colors.

I recommend reading [this](https://discord.com/channels/958989850181259274/1152308222036807823) thread on discord that explains how to change an SVGs color with SVGR.

You can make further changes related to SVGs in a future PR.

DevTaehong commented:
> Looks good!
> 
> When it comes to SVGs. I usually recommended using [SVGR](https://react-svgr.com/) which allows you to import svgs as React components that are optimized. Now for modifying the colors you need to jumps through some hoops but at least you don't need to create multiple svgs of different colors.
> 
> I recommend reading [this](https://discord.com/channels/958989850181259274/1152308222036807823) thread on discord that explains how to change an SVGs color with SVGR.
> 
> You can make further changes related to SVGs in a future PR.

Thanks! will look into it!
